src/api/pkg/config/config.go

```diff
@@ -0,0 +1,65 @@
+package config
+
+import (
+	"github.com/imdario/mergo"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+)
+
+type Configuration struct {
+	Cors struct {
+		AllowedOrigins []string `yaml:"allowed-origins"`
+		AllowedHeaders []string `yaml:"allowed-headers"`
+	}
+}
+
+type ConfigurationWithOverrides map[string]Configuration
+
+const configFile string = "config.yaml"
+
+func currentEnvironment() string {
+	env := os.Getenv("ENVIRONMENT")
+	if env == "" {
+		env = "production"
+	}
+	return env
+}
+
+func readConfigFile() ([]byte, error) {
+	data, err := ioutil.ReadFile(configFile)
+
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func GetConfig() (Configuration, error) {
+	config := ConfigurationWithOverrides{}
+	res := Configuration{}
+
+	data, err := readConfigFile()
+
+	if err != nil {
+		return res, err
+	}
+
+	err = yaml.Unmarshal([]byte(data), &config)
+
+	if err != nil {
+		return res, err
+	}
+
+	res = mergeConfig(config, currentEnvironment())
+
+	return res, nil
+}
+
+func mergeConfig(conf ConfigurationWithOverrides, env string) Configuration {
+	var defaults Configuration = conf["default"]
+	var custom Configuration = conf[env]
```

Comment:
`custom := conf[env]` is more idiomatic wherever possible (leave it up to the compiler to work out the type)